src/code-block/settings.ts

```diff
@@ -0,0 +1,62 @@
+import { App, MarkdownPostProcessorContext, normalizePath, parseYaml } from "obsidian";
+import renderError from "~/code-block/errors";
+
+export interface Settings {
+  path: string;
+  limit: number;
+  recursive: boolean;
+  sort: "asc" | "desc";
+  sortby: "name" | "mtime" | "ctime";
+}
+
+const DEFAULT_SETTINGS: Settings = {
+  path: "",
+  limit: 0,
+  recursive: true,
+  sort: "desc",
+  sortby: "mtime",
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnyObject = { [key: string]: any };
+const lowercaseKeys = (obj: AnyObject, deep = false) =>
+  Object.keys(obj).reduce((acc, key) => {
+    acc[key.toLowerCase()] =
+      deep && typeof obj[key] === "object" ? lowercaseKeys(obj[key]) : obj[key];
+    return acc;
+  }, {} as AnyObject);
```

Comment:
[🤔](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:thinking:) I think you actually don't need this because your settings object is not nested. Just lowercasing the first level of keys should be enough?

Reply:
for completeness, but might remove later